scripts/globals.py

```diff
@@ -2,26 +2,54 @@
 import os
 
 import modules.scripts as scripts
-import modules.shared as shared
+from modules.shared import opts, models_path, cmd_opts
+from modules.cmd_args import parser
 
 
-global hide_nsfw_networks_key, hide_nsfw, network_descriptions_path
+global hide_nsfw
 
 
-hide_nsfw_networks_key = "sd_lora_tagger_hide_nsfw_extra_networks"
-network_descriptions_path = os.path.join(scripts.basedir(), "network_descriptions")
+def out(msg):
+    print(f"SD Lora Tagger: {msg}")
+
+
+""" Paths """
+lora_tagger_dir = scripts.basedir()
+models_dir = models_path
+network_descriptions_path = os.path.join(lora_tagger_dir, "network_descriptions")
 
-print(f"SD Lora Tagger: network_descriptions_path={network_descriptions_path}")
+# Used in accordance to https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/cmd_args.py
+model_dirs = {
+    "Lora": cmd_opts.lora_dir,
+    "LyCORIS": cmd_opts.lyco_dir,
+    "embeddings": cmd_opts.embeddings_dir,
+    "hypernetworks": cmd_opts.hypernetwork_dir,
+    "Stable-diffusion": cmd_opts.ckpt_dir
+}
+
+""" Options and Flags """
+js_overrides = ["sdNextExtraNetworks.js", "automatic1111ExtraNetworks.js"]
+override_before_ui = ["lora_script.py", "lycoris_script.py", "ui_extra_networks.py"]
+using_sd_next = parser.description is not None and parser.description == "SD.Next"
+
+""" Other """
+hide_nsfw_networks_key = "sd_lora_tagger_hide_nsfw_extra_networks"
+networks = {
+    "Lora": ["safetensors"],
+    "LyCORIS": ["safetensors"],
+    "embeddings": ["pt", "safetensors"],
+    "hypernetworks": ["pt"],
```

Comment:
The LyCORIS network should only be added if the extension is installed. Otherwise, all functions using this would have to implement additional failsafes, such as `civitai_utils.add_civitai_tags()`

Reply:
Alright, I’ll put a check in globals to set the network if it’s installed